packages/core/lib/utils/Precision.ts

```diff
@@ -0,0 +1,19 @@
+import BigNumber from 'bignumber.js';
+
+// Refer to https://github.com/discreetlogcontracts/dlcspecs/blob/master/PayoutCurve.md#version-0-payout_function
+
+/**
+ * Extract the precision of a number
+ * @param num
+ * @returns The precision of the number
+ */
+export const getPrecision = (num: BigNumber): number =>
+  num.decimalPlaces(16).abs().modulo(1).shiftedBy(16).toNumber();
```

Comment:
Great question, had to think a bit again about this one. The `getPrecision` and `toPrecision` are methods based on what is defined in the specs [here](https://github.com/discreetlogcontracts/dlcspecs/blob/master/PayoutCurve.md#version-0-payout_function)

As defined in the specs

> Precision is set to be the first 16 bits of the payout after the binary point which were rounded away

So in the case of `BigNumber(1000.1)`, the `precision` value is `1000000000000000` since the only digit after the decimal point is 1 and the number itself can hold up to 16 digits.

Another example is `getPrecision(new BigNumber(1.112233445566778899))` which returns `1122334455667788` which is the first 16 digits after the decimal point.

Reply:
Ah that makes more sense

Here they are referring to `extraPrecision` meaning digits that come after the decimal. Not precision in terms of significant digits. Would be good to add a comment here to clarify.